src/anemoi/training/train/train.py

```diff
@@ -144,10 +144,29 @@ def model(self) -> GraphForecaster:
             "metadata": self.metadata,
             "statistics": self.datamodule.statistics,
         }
+        model = GraphForecaster(**kwargs)
+
         if self.load_weights_only:
             LOGGER.info("Restoring only model weights from %s", self.last_checkpoint)
-            return GraphForecaster.load_from_checkpoint(self.last_checkpoint, **kwargs)
-        return GraphForecaster(**kwargs)
+            checkpoint = torch.load(self.last_checkpoint)
+            checkpoint_state_dict = checkpoint["state_dict"]
```

Comment:
It may be good to have this behaviour to be toggleable, so a config is needed to be set to enable this size checking. Otherwise a user may attempt to load a checkpoint of an invalid model, and every layer just skipped with only logs as awareness.

Reply:
Yes, that makes sense. 

Btw, do we have a check that a checkpoint we specify when warm starting actually exist? at least at some point the model just complaint that it can't find the checkpoint and then does a cold start.

Reply:
I'm not sure, I would imagine that it should fail when it can't find the checkpoint